### DISCUSSION TITLE
Ikea Vindriktning particulate sensor Faceplate with oled

### DISCUSSION THREAD

**Discussion by evilpotato** [arendst/Tasmota#14084] [Show and tell]
Hey all!

I felt like sharing a quick hack I put together using an $11 Ikea Vindriktning, esp8266 with tasmota, and an oled.
I thought having a bit more information than just the LED color would be useful, so I made a 3d model of the faceplate to install an 128x32 oled screen.  I thought it turned out pretty clean, so here ya go!

![front](https://user-images.githubusercontent.com/4966568/146610554-6d2cd625-394b-4995-856c-cdfec6af4306.jpg)

I've uploaded the stl file to [https://www.thingiverse.com/thing:5167213](https://www.thingiverse.com/thing:5167213)
For more info about the interface, see [https://blakadder.com/vindriktning-tasmota/](https://blakadder.com/vindriktning-tasmota/)

**Comment by sfromis** [comment]
If you are keeping that OLED screen on all the time (?), pixel burnout is likely to come....

**Comment by joba-1** [comment]
easy enough to replace, I guess. Nice project

**Comment by barbudor** [comment]
Love it 😍
My Vindxxxxx is still on my desk waiting for cyber improvement

**Comment by plewka** [comment]
Great, this is interesting! 
I was thinking about doing a cylindric baseplate for it and put ESPs there
...no precision required maybe with accu. 
Is there a chance to go "wireless" for a while? 
Anybody checked how much power it takes (without ESP?).

Maybe one could add an IR-based motion detection to protect the display
and add second use. https://www.ebay.de/sch/i.html?_nkw=AM312

**Reply by sfromis** [reply]
As a project upgrade, you could also consider using an e-paper display, instead of those OLED displays not really suitable for always-on.

**Reply by sfromis** [reply]
When it comes to power usage, in a recent test with D1 Mini and one sensor, this alone depleted a "pretty good" 18650 cell in two days. Did nothing to make it as low-power as possible. With a fan always running, and LED lightshow, my guess would be that Vindriktning (horrible name, means "wind direction") lasts less than that.

Since Ikea suggests using "USB-C power adapter (5.0V, 2.0A)" instead of "more basic" 1A or 0.5A, it does indeed not look like low power was a design goal for a device expected to be always plugged in.

**Reply by evilpotato** [reply]
I recently ran across a sale on a [TTGO T-Audio board](https://github.com/LilyGO/TTGO-TAudio) and thought it'd be a good platform to start with for that kind of setup. 
It's cylindrical, has integrated WS2812s, MPU9250 and mics for presence sensing, and a WM8978 if I want to make it more than just an air quality sensor (a Bluetooth speaker would also be welcome wherever I'd want to detect air quality anyways, right?).  A charging circuit is ready to go as well.

So far I've got it hooked up to a pms7003, an scd41, an oled, and an sht30 that's been made redundant with the scd41.  I'm still working out the Tasmota rules for the leds and oled, and the physical design is not quite finished yet, but it's getting there.
![20211221_000356](https://user-images.githubusercontent.com/4966568/146874755-22c40818-d0ad-42e4-9d73-03cda0be5ee5.jpg)

**Comment by joba-1** [comment]
Just noticed: @evilpotato you should fix your units: should be µg/m³ not µm³, I guess.

and don't worry too much about oled lifetime. 
I just recently decommissioned a dm800 hd that had an always on oled display. It had some display problems in the last years, but only minor. It was usable for over 10 years and not the reason for replacement. 
On the other hand, I have some particulate sensors taking measurements every 5 min for some years now and had to replace them after 3-5 years because the readings went down slowly. This was fully according to their specs - particulate sensors wear out after that usage period - so probably faster than the oleds.

Regarding battery usage of ESPs: I wouldn't recommend it. Sleep current is too high for my taste. 
I now use STM32L* low power chips along with a LoRa chip to send (not receive) sensor data based on battery or solar. Receiving side is an ESP with a LoRa chip (good old RFM95W) as a gateway. Sensors are running my own firmware and pcb (see my github if you like) to avoid power consuming functions and components. This way sensor battery lifetime is over a year (two normal AAA batteries going from 3V down to ~2V in that time). If you can live with longer measurement intervals, it can be much longer, because most energy budget is used during transmit and sleep current is negligable compared to that on my devices.

**Reply by sfromis** [reply]
Just to be clear, I did not mean OLED as such, but specifically those small and very cheap chinese OLED displays like ones based on SSD1306. While they are fine for intermittent usage, they are known to have durability issues if always-on. Not horribly bad, just a consideration worth remembering. Random quote from the net:
> One really important caveat: These displays have a finite lifetime. At a full brightness running all the time, they will become noticeably dimmer after about 2 years. And the displays do occasionally fail.

**Reply by joba-1** [reply]
agreed - and it would surely help to move the displayed graphics and text a bit over time. Looks like there is enough horizontal space for that, Maybe as simple as + 1px on every 100th update is good enough already.

**Comment by ComputerZOO** [comment]
Hey, nice project! Would you please share the settings you've made (display the graphics eg)

**Comment by dieskim** [comment]
Are you using a 128x32 or 128x64 display? could you share how you got the 128x32 display to work?
